astropy/table/table.py

```diff
@@ -675,7 +675,7 @@ def _convert_col_for_table(self, col):
         MaskedColumn if the table is masked.  Table subclasses like QTable
         override this method.
         """
-        if isinstance(col, Column) and not col.__class__ is self.ColumnClass:
```

Comment:
Here, I think the test is right: one only wants to select `ColumnClass` subclass instances, not instances of `ColumnClass` itself. Aside: for speed, one probably wants to swap the two phrases `is not` is far faster than `isinstance` and usually `col.__class__` *is* `self.ColumnClass`.